### ISSUE TITLE
DSpace 6 build fails using latest Maven version due to blocking of restlet m2 repository

### ISSUE THREAD

**Issue by cwilper** [DSpace/DSpace#3247]
**Describe the bug**

Using the latest version of maven (3.8.1), building DSpace 6 (or DSpace 5), the build fails with a message like the following (formatted for readability):

```
[ERROR] Failed to execute goal on project dspace-solr: 
  Could not resolve dependencies for project org.dspace:dspace-solr:war:6.4-SNAPSHOT: 
    Failed to collect dependencies at org.apache.solr:solr-core:jar:4.10.4 
      -> org.restlet.jee:org.restlet:jar:2.1.1: 
        Failed to read artifact descriptor for org.restlet.jee:org.restlet:jar:2.1.1: 
          Could not transfer artifact org.restlet.jee:org.restlet:pom:2.1.1 from/to 
          maven-default-http-blocker (http://0.0.0.0/): 
            Blocked mirror for repositories: [
              maven-snapshots (http://oss.sonatype.org/content/repositories/snapshots, default, snapshots), 
              maven-restlet (http://maven.restlet.org, default, releases+snapshots), 
              apache.snapshots (http://repository.apache.org/snapshots, default, disabled)
            ] -> [Help 1]
```

This is failing due to a security measure added to maven 3.8.1 ([see release notes](https://maven.apache.org/docs/3.8.1/release-notes.html)). This measure prevents the use of plain old http maven repositories in order to reduce the chance of man-in-the-middle attacks.

**To Reproduce**

1. Clear out your ~/.m2/repository/org/restlet/ directory, if present (this may not be wholly necessary, but it ensures maven will be forced to try to download the library in any case).
2. Make sure you're using the latest version of maven (currently 3.8.1)
3. Try to build dspace...the build should fail with an error like the above.

**Expected behavior**

It should successfully build, as it does with Maven 3.8.0 and below.

**Comment by cwilper** [comment]
I'm not aware of a way to fix this in code, but I've been able to get around the issue (and successfully build with maven 3.8.1) by putting the following in my `~/.m2/settings.xml`:

```
    <mirrors>
        <mirror>
            <id>maven-restlet-http-unblocker</id>
            <mirrorOf>maven-restlet</mirrorOf>
            <name></name>
            <url>http://maven.restlet.org</url>
            <blocked>false</blocked>
        </mirror>
    </mirrors>
```

This permits the use of http for this one repository, for the current user.

**Comment by tdonohue** [comment]
This sounds related to this StackOverflow question as well. https://stackoverflow.com/questions/66962265/dspace-6-3-solr-webapp-build-error-org-restlet-jee-blocked-mirror

The only solution mentioned there was downgrading to < Maven 3.8.1

**Comment by mwoodiupui** [comment]
According to https://mvnrepository.com/artifact/org.restlet.jee/org.restlet/2.1.1 this artifact should be available at https://repo.spring.io/plugins-release/ so maybe adding that repo. would resolve this?

**Comment by cwilper** [comment]
@tdonohue I came across that also, and that's what I did at first. But then I came across this, which led me to the workaround above: https://stackoverflow.com/questions/67001968/how-to-disable-maven-blocking-external-http-repositores

@mwoodiupui Good thought, and preferable to loading over the old http repo. I tried setting the mirror url to that, but I get "authentication failed". For example: https://repo.spring.io/plugins-release/org/restlet/jee/org.restlet/2.1.1/org.restlet-2.1.1.pom 

I looked around a bit more and found Talend (the company that acquired Restlet in 2017) has a https repository that is public. This worked for me:

```
    <mirrors>
        <mirror>
            <!-- Resolve maven-restlet over https using a newer repository -->
            <id>restlet-https</id>
            <mirrorOf>maven-restlet</mirrorOf>
            <name>Talend Restlet Repository</name>
            <url>https://maven.restlet.talend.com</url>
        </mirror>
    </mirrors>
```
